samples/LiveEffect/src/main/java/com/google/oboe/samples/liveEffect/MainActivity.java

```diff
@@ -130,6 +130,10 @@ public void onClick(View v) {
         LiveEffectEngine.setDefaultStreamValues(this);
         setVolumeControlStream(AudioManager.STREAM_MUSIC);
 
+        if (!isRecordPermissionGranted()){
+            requestRecordPermission();
```

Comment:
Does this block?
If YES then it may cause an ANR.
If NO then you may need code to start the foreground service and test in onRequestPermissionResult().

Reply:
Thanks for bringing this up! I fixed this in the recent commit.